python/graphstorm/graphpartitioning/metadata_schema.py

```diff
@@ -0,0 +1,706 @@
+                  are created, and
+                  ``[file1, file2, ...]`` is a list of files in which features
+                  are stored for this edge type.
+        """
+        self._etype_feature_files = {}
+        edge_data = self._data.get(STR_EDGE_DATA, {})
+        for etype, etype_info in edge_data.items():
+            for feature_name, feature_info in etype_info.items():
+                data_files = feature_info.get(STR_DATA, [])
+                file_type = None
+                delimiter = None
+                if len(data_files) > 0:
+                    file_type = feature_info[STR_FORMAT][STR_NAME]
+                    delimiter = feature_info[STR_FORMAT].get(
+                        STR_FORMAT_DELIMITER, ","
+                    )
+
+                self.__check_ifexists(data_files, f"{etype}/{feature_name}")
+                self._etype_feature_files[(etype, feature_name)] = (
+                    file_type,
+                    delimiter,
+                    data_files,
+                )
+
+    def _init_etype_files(
+        self,
+    ):
+        """Initialization function to build a map for the edge types and the
+        corresponding information about the set of associated files. For
+        instance, etype: (file_type, delimiter, [file1, file2, ...])
+            where ``etype`` is an edge type in the input graph,
+                  ``file_type`` is type of the file, numpy and csv are the
+                  supported file formats,
+                  ``delimiter`` is the delimiter used in these files, and
+                  ``[file1, file2, ...]`` is a list of files in which the
+                  edges are stored for the corresponding edge type.
+        """
+        self._etype_files = {}
+        for etype, etype_info in self._data[STR_EDGES].items():
+            file_type = etype_info[STR_FORMAT][STR_NAME]
+            delimiter = etype_info[STR_FORMAT][STR_FORMAT_DELIMITER]
+            data_files = etype_info[STR_DATA]
+
+            self.__check_ifexists(data_files, etype)
+            self._etype_files[etype] = (file_type, delimiter, data_files)
+
+        assert len(self._etype_files) == len(self._etypes), (
+            f"In the metadata file there are some edges"
+            f" for which there are no corresponding edge files."
+            f" etypes = {self._etypes} are all the edge files. "
+            f" etypes, for which edge files are present: "
+            f" {self.etype_files.keys()}"
+        )
+
+    def _init_ntype_maps(self):
+        """Initialization function to build unique id maps for node types of
+        the input graph. This function creates a list of node types present in
+        the graph, and unique-id <-> node type and reverse maps.
+        """
+        self._ntypes = self._data[STR_NODE_TYPE]
+        self._ntype_id_map = {
+            ntype_name: idx for idx, ntype_name in enumerate(self._ntypes)
+        }
+        self._id_ntype_map = dict(enumerate(self._ntypes))
```

Comment:
the same comment here.

Reply:
Removed for now. If needed will add later that will justify the need at the time.